### PR TITLE
Exception Handling / IT-Academy / JC1-117-22-dev / Haretski Aliaksei / Done tasks 01-07

### PR DESCRIPTION
**Pull request by haretski-aliaksei**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [ ] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/exception/ExceptionTask06CollarIsPutOn.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.rakovets.course.java.core.practice.exception_handling.exception;
+
+public class ExceptionTask06CollarIsPutOn extends RuntimeException{
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/exception/ExceptionTask06CollarIsPutOn.java:3]:
```suggestion
public class ExceptionTask06CollarIsPutOn extends RuntimeException {
```

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public class Task03Demo {
+    public static void main(String[] args) {
+        int testValue = 1;
+
+        Task03 task03 = new Task03();
+
+        try {
+            task03.checkExceptionTask03(testValue);
+        } catch (ExceptionTask03 exception) {
+            logger.info(exception.getMessage());
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java:16]: wrong solution

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+public class Task04Demo {
+    public static void main(String[] args) {
+        String testValue = null;
+
+        Task04 task04 = new Task04();
+
+        try {
+            task04.checkExceptionTask04(testValue);
+        } catch (ExceptionTask04 exception) {
+            logger.info(exception.getMessage());
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task04Demo.java:16]: wrong solution

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task06Dog.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package com.rakovets.course.java.core.practice.exception_handling;
+
+import com.rakovets.course.java.core.practice.exception_handling.exception.ExceptionTask06CollarIsPutOn;
+import com.rakovets.course.java.core.practice.exception_handling.exception.ExceptionTask06LeashIsPutOn;
+import com.rakovets.course.java.core.practice.exception_handling.exception.ExceptionTask06MuzzleIsPutOn;
+
+public class Task06Dog {
+    String name;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task06Dog.java:8]:
```suggestion
    private String name;
```

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import static com.rakovets.course.java.core.practice.exception_handling.JavaUtilLogging.logger;
+
+public class Task06Demo {
+    public static void main(String[] args) {
+        Task06Dog dog = new Task06Dog("Dog", false, false, false);
+
+        try {
+            dog.checkReadinessDogForWalk();
+        } catch (ExceptionTask06CollarIsPutOn | ExceptionTask06LeashIsPutOn exception) {
+            exception.printStackTrace();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java:16]: printStackTrace...
Where?

**Review by rakovets — CHANGES_REQUESTED** [review]
You need to fix and add correct solutions

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/exception/ExceptionTask06CollarIsPutOn.java (hunk 1 above)
Comment (haretski-aliaksei) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/exception/ExceptionTask06CollarIsPutOn.java:3]: fixed

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java (hunk 2 above)
Comment (haretski-aliaksei) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task03Demo.java:16]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task06Dog.java (hunk 4 above)
Comment (haretski-aliaksei) [reply src/main/java/com/rakovets/course/java/core/practice/exception_handling/Task06Dog.java:8]: fixed

File: src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java (hunk 5 above)
Comment (haretski-aliaksei) [reply src/test/java/com/rakovets/course/java/core/practice/exception_handling/Task06Demo.java:16]: fixed

**Comment by haretski-aliaksei** [comment]
Fixed PR: https://github.com/rakovets/course-java-core/pull/1032

**Pull request closed without merging**